src/plugins/dev_tools/public/dev_tools_icon.tsx

```diff
@@ -3,23 +3,117 @@
+  deps: DevToolsSetupDependencies;
+  title: string;
+}) {
+  const [modalVisibile, setModalVisible] = useState(false);
+  const elementRef = useRef<HTMLDivElement | null>(null);
+  const setMountPoint = useCallback((renderFn) => {
+    renderFn(elementRef.current);
+    return () => {};
+  }, []);
 
-export function DevToolsIcon({ core, appId }: { core: CoreStart; appId: string }) {
   return (
-    <EuiButtonIcon
-      aria-label="go-to-dev-tools"
-      iconType="consoleApp"
-      onClick={() => {
+    <>
+      <EuiButtonIcon
+        aria-label="go-to-dev-tools"
+        iconType="consoleApp"
+        onClick={() => {
+          setModalVisible(true);
+        }}
+      />
+      {modalVisibile ? (
         /**
-         * This is a workaround in 2.16, once devTools being refactor to a drawer, we can remove the setCurrentNavGroup line.
+         * We can not use OuiModal component here because OuiModal uses OuiOverlayMask as its parent node
+         * but overlay mask has a default padding bottom that prevent the modal from covering the whole page.
          */
-        core.chrome.navGroup.setCurrentNavGroup(DEFAULT_NAV_GROUPS.dataAdministration.id);
-        core.application.navigateToApp(appId);
-      }}
-    />
+        <EuiOverlayMask className="devToolsOverlayMask" headerZindexLocation="below">
+          <div style={{ width: '100vw', height: '100vh', maxWidth: '100vw' }}>
+            <EuiButtonIcon
+              iconType="cross"
+              onClick={() => setModalVisible(false)}
+              className="euiModal__closeIcon"
+              color="text"
+            />
+            <EuiPanel
+              className="eui-fullHeight"
+              paddingSize="none"
+              hasBorder={false}
+              hasShadow={false}
+            >
+              <EuiFlexGroup direction="column" className="eui-fullHeight devToolsModalContent">
+                <EuiFlexItem grow={false}>
+                  <EuiSpacer />
+                  <EuiFlexGroup justifyContent="spaceBetween">
+                    <EuiFlexItem grow={1}>
+                      <EuiText size="s">
+                        <h2>{title}</h2>
+                      </EuiText>
+                    </EuiFlexItem>
+                    <EuiFlexItem grow={false}>
+                      <div ref={(element) => (elementRef.current = element)} />
+                    </EuiFlexItem>
+                  </EuiFlexGroup>
+                </EuiFlexItem>
+                <EuiFlexItem className="devAppWrapper">
+                  <MainApp
+                    devTools={devTools}
+                    savedObjects={core.savedObjects}
+                    notifications={core.notifications}
+                    dataSourceEnabled={!!deps.dataSource}
+                    dataSourceManagement={deps.dataSourceManagement}
+                    useUpdatedUX
+                    setMenuMountPoint={setMountPoint}
+                    RouterComponent={MemoryRouter}
```

Comment:
This change depends on https://github.com/opensearch-project/dashboards-query-workbench/pull/381.